### ISSUE TITLE
build errors on debian 12 bookworm, gcc 12.2

### ISSUE THREAD

**Issue by bve-ape** [gateway-services/chipdna-winlin-client-cpp#26]
Hi,
I am trying to build the project on debian 12 and I get the following error:
```
../src/parameterset.cpp: In function ‘std::string ChipDnaClientLib::intToString(int)’:
../src/parameterset.cpp:9:80: error: taking address of rvalue [-fpermissive]
    9 |                 return static_cast<std::ostringstream*>(&(std::ostringstream() << number))->str();
      |                                                          ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~

```
Could anyone take a look?
Thanks

**Comment by wunderfeyd** [comment]
I made a pull request in the past. One of the inner commits contains a fix for this (newer compiler versions).

https://github.com/gateway-services/chipdna-winlin-client-cpp/pull/17 (https://github.com/gateway-services/chipdna-winlin-client-cpp/pull/17/commits/5168db5390cdffdddcd969d2001a9581877624d5)

Feel free to use it. ;)

**Comment by cmsws** [comment]
Thank you for the notification, we will upload a fix for this shortly.  

In the meantime we would suggest changing the `CppLib/src/parameterset.cpp` file as in the pull request referenced by @wunderfeyd.

**Comment by bve-ape** [comment]
Thank you for the info!
Regards

**Comment by cmsws** [comment]
Hi bve-ape, 

A fix for this is included in 3.11 Lake Louise.
